modules/jupyter-widget/src/binary-transport.js

```diff
@@ -0,0 +1,117 @@
+import jsonConverter from './create-deck';
+
+// Functions to wrangle data from pydeck's row major order matrix dataframe to the deck.gl binary data format.
+// The format used is described here:
+// https://deck.gl/#/documentation/developer-guide/performance-optimization?section=supply-attributes-directly
+
+// eslint-disable-next-line complexity
+function dtypeToTypedArray(dtype, data) {
+  // Supports converting a numpy-typed array to a JavaScript-typed array
+  // based on a string value dtype and a DataView `data`
+  switch (dtype) {
+    case 'int8':
+      return new Int8Array(data);
+    case 'uint8':
+      return new Uint8Array(data);
+    case 'int16':
+      return new Int16Array(data);
+    case 'uint16':
+      return new Uint16Array(data);
+    case 'float32':
+      return new Float32Array(data);
+    case 'float64':
+      return new Float64Array(data);
+    case 'int32':
+      return new Int32Array(data);
+    case 'uint32':
+      return new Uint32Array(data);
+    case 'int64':
+      return new BigInt64Array(data); // eslint-disable-line no-undef
+    case 'uint64':
+      return new BigUint64Array(data); // eslint-disable-line no-undef
+    default:
+      throw new Error(`Unrecognized dtype ${dtype}`);
+  }
+}
+
+function deserializeMatrix(arr, manager) {
+  /*
+   * Data is sent from the pydeck backend
+   * in the following format:
+   *`
+   * {'<layer ID>':
+   *     {'<accessor function name e.g. getPosition>': {
+   *        {
+   *          layer_id: <duplicate ID of layer as above>
+   *          column_name: <string name of column>
+   *          matrix: {
+   *            data: <binary data, a row major order representation of a matrix>,
+   *            shape: [<height>, <width>],
+   *            dtype: <string representation of typed array data type>
+   *          }
+   *
+   * Multiple layers and multiple accessors are possible.
+   *
+   * Objects at the JSON path `<accessor name>.matrix.data`
+   * are vectors representing a 1 x n matrix,
+   * or they can represent a m x n matrix in row major order form;
+   * shape of the matrix is given at <accessor name>.matrix.data.shape
+   */
+  if (!arr) {
+    return null;
+  }
+  const renderable = {};
+  for (const datum of arr.payload) {
+    // Each entry here represents a single column in
+    // a pandas.DataFrame arriving from the pydeck backend
+    const layerId = datum.layer_id;
+    const accessor = datum.accessor;
```

Comment:
`accessorName`. `accessor` refers to an object of shape `{value, size, stride, offset, ...}`.